### ISSUE TITLE
aria-required-children violation not reported for listbox without children

### ISSUE THREAD

**Issue by afonsoduarte** [dequelabs/axe-core#3409]
Product: axe-core

Expectation: Axe-DevTools and axe-core should report the same two violations for this page https://axe-core-aria-required-children-bug.afonsoduarte.repl.co: one for the listbox with a bad child, one for the listbox without any children.

Actual: axe-core reports only one violation, the listbox with a bad child, but does not report the listbox without any children (check console on https://axe-core-aria-required-children-bug.afonsoduarte.repl.co).

Motivation: A11y reports should be consistent between what QAs see in the browser extension and what we see in automated tests.

<pre><code>
axe-core version: 4.4.1
axe-webdriver, extension or other integration version: 4.4.1
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. An empty listbox [is allowed to be empty](https://github.com/dequelabs/axe-core/blob/develop/doc/check-options.md#aria-required-children) and will return a Needs Review (`incomplete`) result. In the devtools extension, it'll have a flag next to the result to denote that it's Needs Review rather than a violation. 

When I run the extension on the page, I see there are 2 issues, 1 is Needs Review and 1 that is Critical. The critical one is the first "bad listbox child" example while the Needs Review on is that the listbox is empty. 

Are you seeing something different?

<img width="571" alt="image" src="https://user-images.githubusercontent.com/2433219/157718194-a59d2674-6de4-4fe9-a25d-98282b3a3b73.png">

<img width="576" alt="image" src="https://user-images.githubusercontent.com/2433219/157718394-af0821cc-3bdd-41e1-a1cf-4c48e33bbba5.png">

**Comment by afonsoduarte** [comment]
thanks for looking into this @straker, I can confirm I am seeing the same as you. I didn't realise a Need Review issue would appear in `incomplete` in axe-core.
